design/proposals/spike-cli-dockerless.md

```diff
@@ -0,0 +1,209 @@
+# Spike - Containerless CLI
+
+## Table of Contents
+
+[//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
+
+  * [Useful links](#useful-links)
+  * [Background](#background)
+      - [Requirements](#requirements)
+  * [Research](#research)
+      - [Estimation](#estimation)
+      - [Takeways and recommendation](#takeways-and-recommendation)
+  * [Documentation](#documentation)
+  * [Open questions](#open-questions)
+  * [Implementation plan](#implementation-plan)
+
+## Useful links
+
+[//]: # "Add links that may be useful for the reader"
+
+| **Term**   | **Description**                                |
+| ---------- | ---------------------------------------------- |
+| Epic       | https://github.com/cyberark/conjur/issues/1892 |
+| Ruby CLI   | https://github.com/cyberark/conjur-cli         |
+| Go CLI     | https://github.com/cyberark/conjur-cli-go      |
+| Python CLI | https://github.com/cyberark/conjur-api-python3 |
+
+## Background
+
+[//]: # "Give relevant background for the designed feature. What is the motivation for this solution?"
+
+This document is one piece in the bigger piece of the Dockerless Conjur puzzle. We need to be able to deploy Conjur on a VM. This means we will need to figure out a way to deliver a containerless CLI to allow human-users to interface with Conjur.
+
+In this document we will discuss possible options for a containerless CLI and present a decision for which one we will pursue.
+
+#### Requirements
+
+The CLI design should answer the following requirements:
+
+*Core functionality*
+
+1. For each platform, we should support `init`, `authn` (`login`, `authenticate`),  `policy (load)`, `list`, `show, ` `rotate_api_key` (`host` / `user`) and `fetch secrets` functionalities. These functionalities are must-haves but we should not be limited to providing just these.
+
+*Deployment*
+
+1. Provide a manual and automated way to deploy the CLI
+2. Require a maximum of 2 steps
+3. ***Recommended:*** cross-platform deployment (Linux, Windows, macOS) should all have the same flow
+
+## Research
+
+[//]: # "Elaborate on the issue you are writing a solution for"
```

Comment:
needed?

Reply:
those are comments that don't appear in doc if rendered so I left them because they aren't visible. See here https://github.com/cyberark/conjur/blob/e14435868badec5ca5cce3091a16cd1dc3bbb55e/design/proposals/spike-cli-dockerless.md